source/MaterialXGenShader/ShaderStage.h

```diff
@@ -113,7 +113,14 @@ class MX_GENSHADER_API VariableBlock
     ShaderPort* find(const ShaderPortPredicate& predicate);
 
     /// Add a new shader port to this block.
-    ShaderPort* add(const TypeDesc* type, const string& name, ValuePtr value = nullptr);
+    /// @param type The desired shader port type
+    /// @param name The shader port name
+    /// @param value The value to attach to the shader port
+    /// @param shouldWiden When false, an exception is thrown if the type of the existing port with
+    ///   the same name does not match the requested type. When true, the types can mismatch, and the
+    ///   type of any existing port is widened to match the requested type when necessary.
+    /// @return A new shader port, or a pre-existing shader port with the same name.
+    ShaderPort* add(const TypeDesc* type, const string& name, ValuePtr value = nullptr, bool shouldWiden = false);
```

Comment:
Can you do a quick check on the Python wrapper for this class (in `PyMaterialXGenShader`). I didn't see any issue with the change in signature since I don't think it's exposed but best to double-check :). Thanks.

Reply:
I didn't see any mention of `add` in `PyMaterialX/PyMaterialXGenShader/PyShaderStage.cpp`. Could it be that this is not exposed?